File: app/controllers/quotes_controller.rb
class QuotesController < ApplicationController
  def index
    response = HTTParty.get("https://thesimpsonsquoteapi.glitch.me/quotes")
    quote = response[0] # extract the first quote from the response array
    Quote.create(content: quote["quote"], character: quote["character"])
    @quotes = Quote.all
  end
end

File: app/javascript/controllers/quotes_controller.js
import { Controller } from "@hotwired/stimulus"

// Connects to data-controller="quotes"
export default class extends Controller {
  static targets = ["quotes"]
  refresh() {
    fetch("/quotes")
    .then(response => response.json())
      .then(data => {
        this.quotesTarget.innerHTML = "";
        data.forEach(quote => {
          const quoteHtml = `
            <p>${quote.content}</p>
            <p>${quote.character}</p>
          `;
          this.quotesTarget.insertAdjacentHTML("beforeend", quoteHtml);
        });
      });
  }
}
